src/DynamoCore/Models/DynamoModel.cs

```diff
@@ -750,6 +750,17 @@ private void InitializeCustomNodeManager()
                 if (customNodeSearchRegistry.Contains(info.FunctionId))
                     return;
 
+                var elements = SearchModel.SearchEntries.OfType<CustomNodeSearchElement>().Where(
+                                x =>
+                                {
+                                    return string.Compare(x.Path, info.Path, StringComparison.OrdinalIgnoreCase) == 0;
+                                }).ToList();
+                foreach (var element in elements)
+                {
+                    SearchModel.Remove(element);
```

Comment:
Hi, @Randy-Ma . Removing item from `SearchModel` causes SearchVM reorganization. That means every time, when we remove some item from Search and add it back, tree items are reloaded. This causes some bugs. For purpose like yours (when you just need remove item and then return it back) it's created `Update` method.
I suggest next:

``` c#
foreach (var element in elements)
                {
                    customNodeSearchRegistry.Remove(element.ID);
                    element.SyncWithCustomNodeInfo(info);
                    SearchModel.Update(element);
                }
```

Please, try this approach.
And do not forget to write something like:
``` c#
if (elements.Any())
                    return;
```
In this case `SearchModel` won't add new item 2 times.